lib/mayaUsd/render/MaterialXGenOgsXml/LobePruner.cpp

```diff
@@ -288,8 +377,7 @@ bool LobePrunerImpl::getOptimizedNodeCategory(const mx::Node& node, std::string&
     }
 
     if (canOptimize) {
-        ensureLibraryHasOptimizedShader(ndName, flags);
-        nodeCategory = node.getCategory() + "_" + flags;
+        nodeDef = ensureLibraryHasOptimizedShader(ndName, flags);
```

Comment:
[Question] Could `ensureLibraryHasOptimizedShader()` now take a `Material::NodeDefPtr` instead of a name? Internally, it calls 
```C++
const auto originalNodeDef = _library->getNodeDef(nodeDefName.GetString());
```
which should just give us `nd`, so we might be able to pass that directly.

Reply:
Ah, it's also called in `getOptimizedNodeId()` and passing a name might be simpler there.